### PR TITLE
Overhauling the teams documentation

### LINKED ISSUE #2359: 📚 Documentation: The teams documentation is riddled with errors
### 💭 Description

The Teams documentation (https://appwrite.io/docs/client/teams) contains a lot of errors. Here are some examples:

https://appwrite.io/docs/client/teams#teamsCreateMembership: 
> If initiated from server side SDKs, new member will automatically be added to the team.

> While calling from side SDKs the redirect url can be empty string.

https://appwrite.io/docs/client/teams#teamsGetMemberships:
> Get a team members by the team unique ID.

https://appwrite.io/docs/client/teams#teamsDelete:
> Only team owners have write access for this resource.

I think there are some more, but I think those examples are enough to show, that the Teams documentation is in need of an overhaul. Furthermore, is https://appwrite.io/docs/client/teams#teamsUpdateMembershipRoles completely missing a description.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by rdmchr**
## What does this PR do?

I'm rewriting the teams documentation to make it easier to understand and fix mistakes.

## Related PRs and Issues

Closes #2359 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
✔

### CODE REVIEW

**Comment by rdmchr** [comment]
Current progress:

- [x] Create Team
- [x] List Teams
- [x] Get Team
- [x] Update Team
- [x] Delete Team
- [ ] Create Team Membership
- [ ] Update Membership Roles
- [ ] Get Team Memberships
- [ ] Update Team Membership Status
- [ ] Delete Team Membership

File: docs/references/teams/get-team-members.md [hunk 1]
```diff
@@ -1 +1 @@
-Get a team members by the team unique ID. All team members have read access for this list of resources.
\ No newline at end of file
+Lists a teams members using the team's unique ID. All team members have read access to this endpoint.
```
Comment (kodumbeats) [inline docs/references/teams/get-team-members.md:1]:
```suggestion
Lists a team's members using the team's unique ID. All team members have read access to this endpoint.
```

**Review by kodumbeats — CHANGES_REQUESTED** [review]
Apologies for the delay on the review! This is a good first polish :)

I've requested a few specific changes, but I'm seeing some inconsistency across different sections of the docs changes - could you take a second pass at your text while trying to make each endpoint's description match the style of the rest?

A few ideas to think about:
- Which endpoints should have an article to begin their sentences (a/an/the)?
- How does the tense of these docs match in comparison to others?

Let me know once you've had your go, and I'll review it promptly 🙈

**Comment by rdmchr** [comment]
I think I fixed the inconsistencies and the tense also seem right.
Just one question, which do you prefer?

"_In admin mode, [...]_"
"_On admin mode, [...]_"
https://github.com/rdmchr/appwrite/blob/update-teams-doc/docs/references/teams/list-teams.md?plain=1#L3

I feel like the first one sounds better, but that might just be me.

**Review by kodumbeats — CHANGES_REQUESTED** [review]
We're super close! Thanks for all your hard work on these docs changes. I've got just a few more small things, then this PR is good to go 👍🏻 

1. I agree that adding articles at the beginning of these in-line docs is more readable, but I just realized it doesn't match the other docs. If we make this change here, we should make it everywhere - we can do that at a later time all at once, so for now, let's remove the articles.
2. In the markdown docs files, I think `Use this endpoint to` reads as a bit redundant since it's found in the endpoint docs 🤔 Let's remove this phrase as a prefix.

File: app/controllers/api/teams.php [hunk 2]
```diff
@@ -113,9 +113,9 @@
     ->label('sdk.response.code', Response::STATUS_CODE_OK)
     ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
     ->label('sdk.response.model', Response::MODEL_TEAM_LIST)
-    ->param('search', '', new Text(256), 'Search term to filter your list results. Max length: 256 chars.', true)
-    ->param('limit', 25, new Range(0, 100), 'Results limit value. By default will return maximum 25 results. Maximum of 100 results allowed per request.', true)
-    ->param('offset', 0, new Range(0, 2000), 'Results offset. The default value is 0. Use this param to manage pagination.', true)
+    ->param('search', '', new Text(256), 'Search term to filter results. Max length: 256 chars.', true)
+    ->param('limit', 25, new Range(0, 100), 'Limit how many results will be returned. Returns up to 25 results by default. Maximum of 100 results allowed per request.', true)
+    ->param('offset', 0, new Range(0, 2000), 'Use this value to manage pagination. The default value is 0.', true)
```
[the remark this answers is not in the document]
Comment (kodumbeats) [reply app/controllers/api/teams.php:118]:
```suggestion
    ->param('offset', 0, new Range(0, 2000), 'Results offset. The default value is 0. Use this value to manage pagination.', true)
```

File: app/controllers/api/teams.php [hunk 3]
```diff
@@ -534,11 +534,11 @@
     ->label('sdk.response.code', Response::STATUS_CODE_OK)
     ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
     ->label('sdk.response.model', Response::MODEL_MEMBERSHIP_LIST)
-    ->param('teamId', '', new UID(), 'Team unique ID.')
-    ->param('search', '', new Text(256), 'Search term to filter your list results. Max length: 256 chars.', true)
-    ->param('limit', 25, new Range(0, 100), 'Results limit value. By default will return maximum 25 results. Maximum of 100 results allowed per request.', true)
+    ->param('teamId', '', new UID(), 'Unique team ID.')
+    ->param('search', '', new Text(256), 'Search term to filter your results. Max length: 256 chars.', true)
+    ->param('limit', 25, new Range(0, 100), 'Limit how many results will be returned. By default will return a maximum of 25 results. Maximum of 100 results allowed per request.', true)
     ->param('offset', 0, new Range(0, 2000), 'Results offset. The default value is 0. Use this param to manage pagination.', true)
```
Comment (kodumbeats) [inline app/controllers/api/teams.php:540]:
```suggestion
    ->param('offset', 0, new Range(0, 2000), 'Results offset. The default value is 0. Use this value to manage pagination.', true)
```

**Review by kodumbeats — CHANGES_REQUESTED** [review]

**Review by kodumbeats — CHANGES_REQUESTED** [review]

**Review by kodumbeats — APPROVED** [review]
Made a few small tweaks really quickly, and now it LGTM 👍🏻

**Comment by eldadfux** [comment]
@kodumbeats, as part of 0.12, @TorstenDittmann has dropped the use of the term "Unique". The relevant diff: https://github.com/appwrite/appwrite/pull/2478/files - we should use consistent terms now.

**Comment by rdmchr** [comment]
> @kodumbeats, as part of 0.12, @TorstenDittmann has dropped the use of the term "Unique". The relevant diff: https://github.com/appwrite/appwrite/pull/2478/files - we should use consistent terms now.

Should be fixed now.

**Review by kodumbeats — APPROVED** [review]

**Review by eldadfux — APPROVED** [review]

**Comment by eldadfux** [comment]
Waiting for tests to pass.

**Comment by eldadfux** [comment]
Merged! Thank you so much @rdmchr !!

**Pull request merged**